lib/unleash/toggle_fetcher.rb

```diff
@@ -13,10 +14,15 @@ def initialize
       self.toggle_resource = ConditionVariable.new
       self.retry_count = 0
 
-      # start by fetching synchronously, and failing back to reading the backup file.
       begin
-        fetch
+        # if bootstrap configuration is available, initialize. An immediate API read is also triggered
+        if Unleash.configuration.use_bootstrap?
+          bootstrap
+        else
+          fetch
+        end
```

Comment:
We still want to fetch at once, even if we read from the bootstrap config, no?

```suggestion
        # if bootstrap configuration is available, initialize. An immediate API read is also triggered
        bootstrap if Unleash.configuration.use_bootstrap?
        fetch
```

Reply:
If we do this, then we should be able to drop the changes to the `ScheduledExecutor`, no? (regarding `immediate_execution`)

It means it will block until we fetch the data from the bootstrap but that sounds like an acceptable trade-off?

Reply:
So I'm open to suggestion here but as written this won't work - we'll need to spawn off a thread or other concurrency primitive here to not block the caller's thread. Currently the only thing that really does threading is the ScheduledExecutor. Threading is evil and subtle and I'd rather keep it to one place if possible

Reply:
You have a point that we don't want to block on this method if the bootstrap config is provided.

I'm however still convinced we can do this in a better way, but we can leave the improvements for a separate PR.